### PR TITLE
chore: 🤖 initial setup of HDS

### PR DESCRIPTION
**Pull request by randallmorey**
## Description

Initial setup of HDS:
- Adds the HDS dependency to Rose
- Auto-includes CSS import path for tokens, so that consuming applications do not need further build configuration
- Removes existing flight-icons dependency, migrating over to flight-icons provided by HDS
- Removes incompatible `ember-cli-template-lint` dependency from admin and desktop

Note:  consuming UIs like admin and desktop must still explicitly import HDS into their styles:
```
@import '@hashicorp/design-system-components';
```

:rose: [Rose preview](https://boundary-ui-storybook-git-chore-hds-install-hashicorp.vercel.app/?path=/story/rose-anonymous--page)
:technologist: [Admin preview](https://boundary-ui-git-chore-hds-install-hashicorp.vercel.app)
:desktop_computer: [Desktop preview](https://boundary-ui-desktop-git-chore-hds-install-hashicorp.vercel.app/cluster-url)

### CODE REVIEW

File: addons/rose/index.js [hunk 1]
```diff
@@ -41,6 +42,26 @@ module.exports = {
...
   },
 
+  /**
+   * Finds the HDS styles folder and includes it into the running
+   * application's `sassOptions.includePaths`.
+   */
+  includeHDSStyles(app) {
+    // Resolve a path to this addon's style folder `addon/styles/addon-name`,
+    // where addon-name is resolved from package.json
+    const addonPath = require.resolve(this.name);
```
Comment (randallmorey) [inline addons/rose/index.js:52]:
```suggestion

```

File: addons/rose/index.js [hunk 2]
```diff
@@ -41,6 +42,25 @@ module.exports = {
     app.options.sassOptions.includePaths.push(styleTree);
   },
 
+  /**
+   * Finds the HDS styles folder and includes it into the running
+   * application's `sassOptions.includePaths`.
+   */
+  includeHDSStyles(app) {
+    // Resolve a path to this addon's style folder `addon/styles/addon-name`,
+    // where addon-name is resolved from package.json
```
Comment (randallmorey) [inline addons/rose/index.js:51]:
```suggestion

```

**Review by dvonanderson — APPROVED** [review]
Awesome can't wait

**Review by DhariniJeeva — APPROVED** [review]

File: addons/rose/index.js [hunk 3]
```diff
@@ -41,6 +42,23 @@ module.exports = {
...
+
+    // Setup default sassOptions on the running application
+    app.options.sassOptions = app.options.sassOptions || {};
+    app.options.sassOptions.includePaths =
+      app.options.sassOptions.includePaths || [];
+
+    // Include the addon styles
+    app.options.sassOptions.includePaths.push(stylePath);
+  },
+
```
Comment (calcaide) [inline addons/rose/index.js:61]: This file still having `includeFlightIcons` function. Assuming the answer to [this question](https://github.com/hashicorp/boundary-ui/pull/1388/files#r1003360249) is a YES. 

Do we still need this function to include flight icons within the app?

**Review by calcaide — APPROVED** [review]
Awesome work!! First stone for Boundary UI embracing HDS done!!

`Note: consuming UIs like admin and desktop must still explicitly import HDS into their styles:` I say AMEN to this!

Left to non blocking comments/questions.

**Review by ZedLi — APPROVED** [review]

**Review by cameronperera — APPROVED** [review]

File: addons/rose/index.js (hunk 3 above)
Comment (DhariniJeeva) [reply addons/rose/index.js:73]: We still need this function as this function is for rose icons that use ember inline SVG to display. Discussion re this [here](https://hashicorp.slack.com/archives/C012GTH1C4E/p1666739607931999)

**Pull request merged**